pkg/ipfs/node.go

```diff
@@ -83,6 +83,9 @@ func NewNodeWithConfig(ctx context.Context, cm *system.CleanupManager, cfg types
 
 	// TODO if cfg.PrivateInternal is true do we actually want to connect to peers?
 	if err = connectToPeers(ctx, api, ipfsNode, cfg.GetSwarmAddresses()); err != nil {
+		if err == context.Canceled {
+			return nil, nil // context was canceled, so we should just stop running
+		}
```

Comment:
I guess we could.

Reply:
Done in 194d4f2e2

Reply:
can we do the same for the node info publisher as well :)?

The reason I am asking for the logging message is that it will be very difficult to figure out why things are not working as expected if we unintentionally cancel the context

Reply:
Yes of course, makes a lot of sense. Done in 1d649b1c4